app/events/domains/proxy_domains_changed_event.rb

```diff
@@ -33,6 +33,7 @@ def after_commit
   end
 
   def self.valid?(proxy)
-    !!proxy # TODO: check if deployment_option or domains actually changed
+    # TODO: check if deployment_option or domains actually changed
+    proxy && proxy.provider&.persisted? # Proxy's service or account may have been deleted
```

Comment:
```suggestion
    proxy && proxy.provider.persisted? # Proxy's service or account may have been deleted
```

Small suggestion :smile: Because if `proxy.provider` is `nil`, `nil.persisted?` will be `false`.